benchmarks/mp_pingpong_kernel_stream_latency.cu

```diff
@@ -200,36 +214,208 @@ void wait_recv (int batch_index)
+    CUDA_CHECK(cudaMemset((void *)in, 1, size));
+    CUDA_CHECK(cudaMemset((void *)out, 1, size));
+    CUDA_CHECK(cudaMemset((void *)sindex_d, 0, sizeof(int)));
+    CUDA_CHECK(cudaMemset((void *)windex_d, 0, sizeof(int)));
+    CUDA_CHECK(cudaMemcpyToSymbol(request_count, (void *)&count, sizeof(int), 0, cudaMemcpyHostToDevice));
+
+    CUDA_CHECK(cudaGraphCreate(&graph, 0));
+
+    waitParams.func = (void*)wait_op_kernel;
+    waitParams.gridDim = 1;
+    waitParams.blockDim = 1;
+    waitParams.sharedMemBytes = 0;
+    void *waitArgs[2] = {(void*)&wdesc, (void *)&windex_d};
+    waitParams.kernelParams = waitArgs;
+    waitParams.extra = NULL;
+
+    sendParams.func = (void*)send_op_kernel;
+    sendParams.gridDim = 1;
+    sendParams.blockDim = 1;
+    sendParams.sharedMemBytes = 0;
+    void *sendArgs[2] = {(void*)&sdesc, (void *)&sindex_d};
+    sendParams.kernelParams = sendArgs;
+    sendParams.extra = NULL;
+
+    const int nblocks = over_sub_factor * gpu_num_sm;
+    const int nthreads = 32*2;
+    int n = kernel_size / sizeof(float);
+
+    calcKernelParams.func = (void*)calc_kernel;
+    calcKernelParams.gridDim = 1;
+    calcKernelParams.blockDim = 1;
+    calcKernelParams.sharedMemBytes = 0;
+    void *calcKernelArgs[4] = {(void*)&n, (void *)&value, (void *)&int, (void *)&out};
+    calcKernelParams.kernelParams = calcKernelArgs;
+    calcKernelParams.extra = NULL;
+
+    dummyKernelParams.func = (void*)dummy_kernel;
+    dummyKernelParams.gridDim = 1;
+    dummyKernelParams.blockDim = 1;
+    dummyKernelParams.sharedMemBytes = 0;
+    void *dummyKernelArgs[1] = {(void*)&kernel_size};
+    dummyKernelParams.kernelParams = dummyKernelArgs;
+    dummyKernelParams.extra = NULL;
+
+    if (!my_rank) { 
+        CUDA_CHECK(cudaGraphAddKernelNode(&waitNode, graph, NULL, 0, &waitParams));
+        nodeDependencies.clear();
+        nodeDependencies.push_back(waitNode);
+    } else {
+        CUDA_CHECK(cudaGraphAddKernelNode(&sendNode, graph, NULL, 0, &sendParams));
+        nodeDependencies.clear();
+        nodeDependencies.push_back(sendNode);
+    }
+
+    if (kernel_size > 0) {
+       if (use_calc_size > 0) {
+           CUDA_CHECK(cudaGraphAddKernelNode(&kernelNode, graph, nodeDependencies.data(), nodeDependencies.size(), &calcKernelParams));
+       } else { 
+           CUDA_CHECK(cudaGraphAddKernelNode(&kernelNode, graph, nodeDependencies.data(), nodeDependencies.size(), &dummyKernelParams));
+       }
+       nodeDependencies.clear();
+       nodeDependencies.push_back(kernelNode);
+    } else {
+        nodeDependencies.clear();
+    }
+
+    if (!my_rank) { 
+        CUDA_CHECK(cudaGraphAddKernelNode(&sendNode, graph, nodeDependencies.data(), nodeDependencies.size(), &sendParams));
+    } else { 
+        CUDA_CHECK(cudaGraphAddKernelNode(&waitNode, graph, nodeDependencies.data(), nodeDependencies.size(), &waitParams));i
+    }
+
+    CUDA_CHECK(cudaGraphInstantiate(&graphExec, graph, NULL, NULL, 0));
```

Comment:
use sub-graphs to unroll multiple iterations 
is better than using the iterator/conditional graph node